### PR TITLE
Changes to Postgres binder to prevent SQL injection

### PR DESCRIPTION
**Pull request by mcarolan**
This PR [implements #170]

TL; DR: the main changes made are:
- Removal of SQL injection for Postgres in binder names and keys
- Generalisation of binder tests to allow execution against Postgres binder stores as well as file binder stores
-  Removed schema creation (DDL running) responsibilities from PostgresBinder and PostgresBinderStore :- this allows mewbase to be run with lesser privilege where DDL is disallowed
    - DDL for Postgres is in `src/main/resources/postgres-schema.sql`. Binder tests run using this schema.

These aren't small changes to the Postgres binder/store, and definitely break back compatibility. So see this PR as more of a starter for a discussion.

TS; RE: some more detail:

## Removal of SQL injection for Postgres in binder names and keys
This was implemented by having the Postgres binder/store only use 2 tables:

- `mewbase_binder` :- where metadata of binders (e.g. names) is stored
- `mewbase_binder_data` :- where data is stored per key per binder

As there is no longer a Postgres table per binder, all potentially user provided (e.g. binder name / key) values appear in condition position in queries, so a `PreparedStatement` can be used

## Generalisation of binder tests to allow execution against Postgres binder stores as well as file binder stores
When executed tests run against a real Postgres database, but after
```
SET search_path TO pg_temp
```
has been set. This means all DDL and other queries to run inside the `pg_temp` schema, which is isolated to the connection :- ensuring test isolation.

**Note** Postgres binder store tests are currently not enabled (see `data()` in `BinderTests`). If we're happy to have a docker-compose / a standard agreed location for a Postgres to use for integration tests then we could enable them for every test run

### CODE REVIEW

**Comment by mcarolan** [comment]
( one of the build jobs broke with a `ConnectException`, others passed. could be flaky travis? )

**Comment by NigeWarren** [comment]
@mcarolan Thanks Martin - great stuff.

When  you say "These aren't small changes to the Postgres binder/store, and definitely break back compatibility. So see this PR as more of a starter for a discussion."

Do you mean that the SPI/API is changed or that the database structure has changed and will need migration for anyone running a current database setup ? 

@kings13y Reran Travis and it worked across the whole test matrix

**Comment by mcarolan** [comment]
Ah yep good point @NigeWarren 
No breaking SPI/API changes only thing changed in public interface is extension of `AutoCloseable`
But yes, database structure has changed, so migration will be necessary for existing setups
